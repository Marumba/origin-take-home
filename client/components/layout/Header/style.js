import styled from '@emotion/styled';
import { css } from '@emotion/react';

export const Header = styled.header`
	background: #fff;

	display: flex;
	justify-content: left;
	align-items: center;

	padding: 12px;

	transition: padding ${({ theme }) => theme.animations.durations.fastest} ease-in;

	@media ${({ theme }) => theme.medias.desktop} {
		padding: 24px;
	}
`;

export const originLogo = theme => css`
	height: 24px;

	transition: height ${theme.animations.durations.fastest} ease-in;

	@media ${theme.medias.desktop} {
		height: 32px;
	}
`;
