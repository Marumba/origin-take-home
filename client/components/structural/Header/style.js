import styled from '@emotion/styled';
import { css } from '@emotion/react';

const transitionTime = '200ms';

export const Header = styled.header`
	background: #fff;

	display: flex;
	justify-content: left;
	align-items: center;

	padding: 12px;

	transition: padding ${transitionTime} ease-in;

	@media ${({ theme }) => theme.medias.desktop} {
		padding: 24px;
	}
`;

export const originLogo = theme => css`
	height: 24px;

	transition: height ${transitionTime} ease-in;

	@media ${theme.medias.desktop} {
		height: 32px;
	}
`;
