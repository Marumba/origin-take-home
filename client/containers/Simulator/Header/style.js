import styled from '@emotion/styled';

export const Header = styled.div`
	display: flex;
	align-items: center;
	justtify-content: left;

	svg {
		margin-right: 15px;
	}

	${({ customCss, theme }) => (typeof customCss === 'function' ? customCss(theme) : customCss)}
`;

export const TextWrapper = styled.div`
	display: flex;
	align-items: left;
	justify-content: center;
	flex-direction: column;
`;
export const Title = styled.h2`
	${({ theme }) => theme.mixins.headingSmall};

	display: block;
`;
export const Subtitle = styled.h3`
	display: block;
`;
