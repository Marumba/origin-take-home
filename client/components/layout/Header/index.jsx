import React from 'react';

import { OriginLogo } from '~/components/minimal';

import * as S from './style';

function Header() {
	return (
		<S.Header>
			<OriginLogo customCss={S.originLogo} type="withLink" />
		</S.Header>
	);
}

export default Header;
