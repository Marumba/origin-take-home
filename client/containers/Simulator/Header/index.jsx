import React from 'react';
import { string, customCssType, oneOfType, element } from '~/types';

import * as S from './style';

function Header({ customCss, icon, subtitle, title }) {
	return (
		<S.Header data-testid="simulator-header" customCss={customCss}>
			{icon}
			<S.TextWrapper>
				<S.Title className="highEmphasis">{title}</S.Title>
				<S.Subtitle className="lowEmphasis">{subtitle}</S.Subtitle>
			</S.TextWrapper>
		</S.Header>
	);
}

Header.propTypes = {
	customCss: customCssType,
	icon: oneOfType([string, element]),
	subtitle: string,
	title: string
};

Header.defaultProps = {
	customCss: undefined,
	icon: undefined,
	subtitle: '',
	title: ''
};

export default Header;
