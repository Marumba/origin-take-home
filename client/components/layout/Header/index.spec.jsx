import React from 'react';

import { render, screen } from '~/helpers/testing.customRtl';

import Header from '.';

describe('Component Header', () => {
	it('should render the Origin logo', () => {
		const { container } = render(<Header />);
		expect(screen.getByTitle(/origin home/i)).toBeInTheDocument();
		expect(screen.getByAltText(/origin logo/i)).toHaveStyleRule('height', '24px');

		expect(container.firstChild).toMatchSnapshot();
	});
});
