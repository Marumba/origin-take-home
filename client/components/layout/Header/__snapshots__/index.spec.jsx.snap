// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Header should render the Origin logo 1`] = `
.emotion-0 {
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: left;
  -ms-flex-pack: left;
  -webkit-justify-content: left;
  justify-content: left;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 12px;
  -webkit-transition: padding 100ms ease-in;
  transition: padding 100ms ease-in;
}

@media screen and (min-width: 992px) {
  .emotion-0 {
    padding: 24px;
  }
}

.emotion-2 {
  display: block;
}

.emotion-4 {
  display: block;
  height: 24px;
  -webkit-transition: height 100ms ease-in;
  transition: height 100ms ease-in;
}

@media screen and (min-width: 992px) {
  .emotion-4 {
    height: 32px;
  }
}

<header
  class="emotion-0 emotion-1"
>
  <a
    class="emotion-2 emotion-3"
    href="/"
    title="Origin Home"
  >
    <img
      alt="Origin Logo"
      class="emotion-4 emotion-5"
      src="SvgrURL"
    />
  </a>
</header>
`;
